src/core/search/search_parser_test.cc

```diff
@@ -58,6 +75,18 @@ class SearchParserTest : public ::testing::Test {
     ASSERT_TRUE(caught);                      \
   }
 
+#define CHECK_ALL(...)                                                    \
+  {                                                                       \
+    for (auto input : {__VA_ARGS__})                                      \
```

Comment:
Not a big deal, but I'd prefer to see separate check statements for each input in the test itself. I would be easier to debug as well (we get the line number, and no macros to manually unroll :)
You could, if you want to, add a custom matcher to print the DebugExpr(), but I actually think that it's not as useful if we remove the macros.

Reply:
The line number is preserved, that's why I use a macro. You have all the info you need.

> DebugExpr(), but I actually think that it's not as useful if we remove the macros.

DebugExpr is always useful because it shows you the tree built by the parser